zipline/utils/api_utils.py

```diff
@@ -0,0 +1,124 @@
+# Copyright 2015 Quantopian, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from six import iteritems
+from toolz import valmap
+
+from zipline.utils.preprocess import preprocess
+
+
+def ensure_upper_case(func, argname, arg):
+    if isinstance(arg, str):
+        return arg.upper()
+    else:
+        raise TypeError(
+            "{0}() expected argument '{1}' to"
+            " be a string, but got {2} instead.".format(
+                func.__name__, argname, arg,)
+        )
+
+
+def expect_types(*_pos, **named):
+    """
+    Preprocessing decorator that verifies inputs have expected types.
+
+    Usage
+    -----
+    >>> @expect_types(x=int, y=str)
+    ... def foo(x, y):
+    ...    return x, y
+    ...
+    >>> foo(2, '3')
+    (2, '3')
+    >>> foo(2.0, '3')
+    Traceback (most recent call last):
+       ...
+    TypeError: foo() expected an argument of type 'int' for argument 'x', but got float instead.  # noqa
+    """
+    if _pos:
+        raise TypeError("expect_types() only takes keyword arguments.")
```

Comment:
Would defining `expect_types` as `def expect_types(**named):` have the same effect as our check here?